src/pages.c

```diff
@@ -42,6 +42,53 @@ thp_mode_t init_system_thp_mode;
 /* Runtime support for lazy purge. Irrelevant when !pages_can_purge_lazy. */
 static bool pages_can_purge_lazy_runtime = true;
 
+#ifdef JEMALLOC_PURGE_MADVISE_DONTNEED_ZEROS
+static int madvise_MADV_DONTNEED_zeros_pages = -1;
```

Comment:
Rather than having an ifdef and a runtime flag with the same name but different semantics, it would be better to have (e.g.) `static bool madvise_dont_need_is_faulty` or something.

Reply:
BTW this is not new macros, it is used in code to decide "does madvise zero page?", so I guess it is fine to keep it

Reply:
Yeah, the macro is fine as is I think (I guess arguably it should be renamed to JEMALLOC_PURGE_MADVISE_DONTNEED_SHOULD_ZERO or something); it's the pseudo-shadowing I think is confusing. Really, the macro describes the documented behavior of the system, and the bool describes whether or not the actual behavior conforms.

Reply:
> and the bool describes whether or not the actual behavior conforms.

Yep, I was talking just about macro, option added - `opt_dont_trust_madvise`